iceoryx_posh/CMakeLists.txt

```diff
@@ -135,16 +135,17 @@ add_library(iceoryx_posh
     source/runtime/ipc_runtime_interface.cpp
     source/runtime/ipc_message.cpp
     source/runtime/port_config_info.cpp
-    source/runtime/posh_runtime.cpp
-    source/runtime/posh_runtime_impl.cpp
-    source/runtime/posh_runtime_single_process.cpp
+    source/runtime/posh_runtime.cpp                #
+    source/runtime/posh_runtime_impl.cpp           # @todo iox-#590 These files should go into a separate library iceoryx_posh_runtime
+    source/runtime/posh_runtime_single_process.cpp #
```

Comment:
I'm not quite sure if it makes sense to have a separate `iceoryx_posh_runtime` library. It most probably won't have many files and it is still needed by the `RouDiEnvironment`. I guess we will only know once we tried it.

Reply:
I'll try thtat as a follow-up to this PR.